### PR TITLE
Improve intel 1 to implement 2

### PR DESCRIPTION
**Pull request by kleberbaum**
**Checklist:**

 
 - [x] Have you added an explanation of what your changes do and why you'd like them to be included?
 - [ ] Have you updated or added documentation for the change?
 - [x] Have you tested your changes with successful results?

**Type of Changes:**

 
 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [ ] New feature (non-breaking change which adds functionality)
 - [x] Breaking change (fix or feature that would cause existing functionality to change)

**What is the current behavior? (link to any open issues here)**
- Currently there is no repoUrl in the repository table. #18 
- Currently there is no way to create repositories without relations. #19 
- Currently it is not possible to get data of user, repositories, organisations, languages, calendar, statistic and contribution types from the database. #53 
- Currently the generator weights a bit off. #54 
- Currently the busiest day is stored in a database table. #55 

**What is the new behavior (if this is a feature change)?**
- Now there is a repoUrl in the repository table. #18 
- Now there is a way to create repositories without relations. #19 
- Now it is possible to get data of user, repositories, organisations, languages, calendar, statistic and contribution types from the database. #53 
- Now the generator weights has been adjusted. #54 
- Now the busiest day is calculated from calendar days. #55 

**Other information**:
Ref: https://github.com/agershun/alasql
Ref: #42 The quality of the branch implement-1 is not sufficient.

### CODE REVIEW

File: src/intel/Database/Converter/index.js [hunk 1]
```diff
@@ -0,0 +1,377 @@
...
+    }
+
+    let streak = {};
+    streak.startDate = stat.stsDate;
+    streak.endDate = stat.steDate;
+    streak.total = stat.stTotal;
+    statistics[stat.sYear].streaks.push(streak);
+  
+    // To calculate the no. of days between two dates 
+    const dateDiff = (date1, date2) => Math.abs(Math.ceil((date2.getTime() - date1.getTime()) / (1000 * 3600 * 24)));
```
Comment (schettn) [inline src/intel/Database/Converter/index.js]: The max line length is 119 characters.

File: src/intel/Database/Converter/index.js [hunk 2]
```diff
@@ -0,0 +1,377 @@
...
+    if(!contribTypes.platform[contribPerPlatform.pName]){
+      let platform = {};
+      platform.share = 0;
+      platform.count = 0;
+
+      contribTypes.platform[contribPerPlatform.pName] = platform;
+    }
+
+    contribTypes.platform[contribPerPlatform.pName].count++;
+    contribTypes.platform[contribPerPlatform.pName].share = Math.round(contribTypes.platform[contribPerPlatform.pName].count / contribPerPlatforms.length * 100 * 100) / 100;
```
Comment (schettn) [inline src/intel/Database/Converter/index.js]: The max line length is 119 characters.

File: src/intel/Database/Converter/index.js [hunk 3]
```diff
@@ -0,0 +1,377 @@
...
+      let platform = {};
+      platform.share = 0;
+      platform.count = 0;
+      platform.name = contribPerPlatform.pName;
+
+      contribTypes.user[contribPerPlatform.pId] = platform;
+    }
+
+    contribTypes.user[contribPerPlatform.pId].count++;
+    contribTypes.user[contribPerPlatform.pId].share = Math.round(contribTypes.user[contribPerPlatform.pId].count / contribPerPlatforms.length * 100 * 100) / 100;
```
Comment (schettn) [inline src/intel/Database/Converter/index.js]: The max line length is 119 characters.

File: src/intel/Database/Converter/index.js [hunk 4]
```diff
@@ -0,0 +1,377 @@
...
+      type.share = 0;
+      type.count = 0;
+      type.platform = {};
+      type.user = {};
+
+      contribTypes.contribs[contribPerPlatform.cType] = type;
+    }
+    
+    contribTypes.contribs[contribPerPlatform.cType].count++;
+    contribTypes.contribs[contribPerPlatform.cType].share = Math.round(contribTypes.contribs[contribPerPlatform.cType].count / contribPerPlatforms.length * 100 * 100) / 100;
```
Comment (schettn) [inline src/intel/Database/Converter/index.js]: The max line length is 119 characters.

File: src/intel/Database/Converter/index.js [hunk 5]
```diff
@@ -0,0 +1,377 @@
...
+    if(!contribTypes.contribs[contribPerPlatform.cType].platform[contribPerPlatform.pName]){
+      let platform = {};
+      platform.share = 0;
+      platform.count = 0;
+
+      contribTypes.contribs[contribPerPlatform.cType].platform[contribPerPlatform.pName] = platform;
+    }
+
+    contribTypes.contribs[contribPerPlatform.cType].platform[contribPerPlatform.pName].count++;
+    contribTypes.contribs[contribPerPlatform.cType].platform[contribPerPlatform.pName].share = Math.round(contribTypes.contribs[contribPerPlatform.cType].platform[contribPerPlatform.pName].count / contribPerPlatforms.filter(elem => {
```
Comment (schettn) [inline src/intel/Database/Converter/index.js]: The max line length is 119 characters.

File: src/intel/Database/Converter/index.js [hunk 6]
```diff
@@ -0,0 +1,377 @@
...
+      let platform = {};
+      platform.share = 0;
+      platform.count = 0;
+      platform.name = contribPerPlatform.pName;
+
+      contribTypes.contribs[contribPerPlatform.cType].user[contribPerPlatform.pId] = platform;
+    }
+    
+    contribTypes.contribs[contribPerPlatform.cType].user[contribPerPlatform.pId].count++;
+    contribTypes.contribs[contribPerPlatform.cType].user[contribPerPlatform.pId].share = Math.round(contribTypes.contribs[contribPerPlatform.cType].user[contribPerPlatform.pId].count / contribPerPlatforms.filter(elem => {
```
Comment (schettn) [inline src/intel/Database/Converter/index.js]: The max line length is 119 characters.

File: src/intel/Database/Converter/index.js [hunk 7]
```diff
@@ -0,0 +1,377 @@
...
+      return elem.cType === contribPerPlatform.cType;
+    }).length * 100 * 100) / 100 ;
+
+      //contribTypes.platform[contribPerPlatform.pName] = platform
```
Comment (schettn) [inline src/intel/Database/Converter/index.js]: Please remove comment out code.

File: src/intel/Database/Statements/Select/index.js [hunk 8]
```diff
@@ -0,0 +1,178 @@
...
+FROM
+  statistic
+JOIN
+  streak
+    ON statistic.id = streak.statistic_id
+`;
+
+export const contribPerPlatform = `
+SELECT
+contrib.id as cId,
```
Comment (schettn) [inline src/intel/Database/Statements/Select/index.js]: Please use 2 indents.

File: src/intel/Database/Statements/Select/index.js [hunk 9]
```diff
@@ -0,0 +1,178 @@
...
+export const busiestDay = `
+SELECT
+  date,
+  total,
+  YEAR(date) as year
+FROM
+  calendar
+WHERE
+  total in (
+    SELECT max(c2.total) as total, YEAR(date)
```
Comment (schettn) [inline src/intel/Database/Statements/Select/index.js]: Please use multiple lines in the SQL statement.

File: src/intel/Database/Converter/index.js [hunk 10]
```diff
@@ -0,0 +1,377 @@
...
+      let slice = {};
+      slice.name = languageSlice.sName;
+      slice.color = languageSlice.sColor;
+      slice.size = languageSlice.sSize;
+      slice.share = Math.round(slice.size / totalSize * 100 * 100) / 100;
+
+      pie.slices[languageSlice.sName] = slice;
+    } else {
+      pie.slices[languageSlice.sName].size += languageSlice.sSize;
+      pie.slices[languageSlice.sName].share = Math.round(pie.slices[languageSlice.sName].size / totalSize * 100 * 100) / 100;
```
Comment (schettn) [inline src/intel/Database/Converter/index.js]: The max line length is 119 characters.

File: src/App.test.js [hunk 11]
```diff
@@ -1,4 +1,18 @@
 import * as intel from "./intel";
 
-intel.fill("pinterid");
-//console.log(intel.templateGetPlatform());
+intel.fill({
```
Comment (schettn) [inline src/App.test.js]: The whole intel.fill block should be commented out due to it is just a "how to use".

Comment (Aichnerc) [reply src/App.test.js]: Ah. That makes sense.

**Pull request merged**